docs/developer/timeout_lock.rst

```diff
@@ -0,0 +1,52 @@
+.. highlight:: python
+
+.. _timeout_lock:
+
+TimeoutLock
+============
+
+Overview
+---------
+
+In OCS, operations are mainly asynchronous, however most agents have
+restrictions on which operations are allowed to be run simultaneously.
+For instance, when only one operation can query with a hardware device at a time
+through serial communication, and so trying to control hardware variables while
+an acquisition process is in progress is dangerous.
+
+In these cases, it is required for agents to create a `lock` which restricts
+what operations can run simultaneously. For instance, in your agent it may
```

Comment:
Another wording suggestion for the sentence that starts with "For instance..." -> "In our example, it is necessary for each operation that communicates over with the hardware device to acquire the lock."